packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -24,6 +24,19 @@ const double _kFloatingActionButtonMargin = 16.0; // TODO(hmuller): should be de
 const Duration _kFloatingActionButtonSegue = const Duration(milliseconds: 200);
 final Tween<double> _kFloatingActionButtonTurnTween = new Tween<double>(begin: -0.125, end: 0.0);
 
+/// Returns a path for a notch in the outline of a shape.
+///
+/// The path makes a notch in the host shape that can contain the guest shape.
+///
+/// The [host] is the bounding rectangle for the host shape, and [guest] is the
+/// bounding rectangle for the guest shape.
+///
+/// The [start] and [end] arguments are points on the outline of the host shape
+/// that will be connected by the returned path.
+///
+/// The returned path might pass inside the content bounds area.
+typedef Path NotchMaker(Rect host, Rect guest, Offset start, Offset end);
```

Comment:
I'm surprised by this signature. I would have expected the NotchMaker to take a Path, not return one.

Reply:
It felt cleaner to me to just return a Path segment instead of mutating an argument.
When the BAB is building it's outline path it is doing:
```..extendWithPath(notchMaker(...), Offset.zero) //...```